src/main/java/io/reactivex/exceptions/CompositeException.java

```diff
@@ -278,7 +278,7 @@ public int size() {
      * @param e the {@link Throwable} {@code e}.
      * @return The root cause of {@code e}. If {@code e.getCause()} returns {@code null} or {@code e}, just return {@code e} itself.
      */
-    private Throwable getRootCause(Throwable e) {
+    /*private */Throwable getRootCause(Throwable e) {
```

Comment:
That `cause == root` case is borderline impossible, unless this method gets called with `this`.